Issue compiling minio on Smartos/Illumos

Hi all,

I am trying to compile minio on smartos but i am getting the following error when i run following command:
go get -u github.com/minio/minio
then after i while i get following error:
github.com/minio/minio/cmd
root/go/src/github.com/minio/minio/cmd/fs-v1-helpers.go:270:17: undefined: lock.Open

The go language that i am using is:
go version go1.10 solaris/amd64

Anybody any ideas as i have looked on the internet to found i solution but no luck
REgards
Filip

--- a/pkg/lock/lock_solaris.go
+++ b/pkg/lock/lock_solaris.go
@@ -98,3 +98,8 @@ func TryLockedOpenFile(path string, flag int, perm os.FileMode) (*LockedFile, er
 func LockedOpenFile(path string, flag int, perm os.FileMode) (*LockedFile, error) {
 	return lockedOpenFile(path, flag, perm, syscall.F_SETLKW)
 }
+
+// Open - Call os.OpenFile
+func Open(path string, flag int, perm os.FileMode) (*os.File, error) {
+	return os.OpenFile(path, flag, perm)
+}

